src/scipp/io/hdf5.py

```diff
@@ -176,6 +177,17 @@ def _data_handler_lut():
     return handler
 
 
+def _write_unit_attr(dset, unit):
+    unit_dict = unit.to_dict()
+    dtype = [(key, type(val)) for key, val in unit_dict.items()]
+    dset.attrs.create('unit', tuple(unit_dict.values()), shape=(), dtype=dtype)
+
+
+def _read_unit_attr(ds):
+    u = ds.attrs['unit']
+    return Unit.from_dict({name: u[name] for name in u.dtype.names})
```

Comment:
This is not checking the version? You rely on key order being unchanged?

Reply:
The version is checked by `from_dict`. The key order should not matter here. Everything is looked up by name.

Reply:
Hmm, I think I missed how the names are stored in the dtype...? This looks confusing.

Reply:
It is a regular structured dtype https://numpy.org/doc/stable/user/basics.rec.html#structured-datatype-creation

Here, `u` is a numpy array with a dtype like `[('__version__', int), ('meter', int), ('second', int), ('multiplier', float)]`. So `u.dtype.names` is `('__version__', 'meter', 'second', 'multiplier')`. And `u` can be indexed like a dict with those names.

Reply:
And h5py translates between those dtypes and compound types in HDF5.